src/core/ActivityInfoTree.cpp

```diff
@@ -272,4 +275,73 @@ void ActivityInfoTree::init()
     qmlRegisterSingletonType<QObject>("GCompris", 1, 0, "ActivityInfoTree", menuTreeProvider);
     qmlRegisterType<ActivityInfo>("GCompris", 1, 0, "ActivityInfo");
 }
+void ActivityInfoTree::beginsearch()
+{
+
+     m_menuTree = m_searched;
+     emit menuTreeChanged();
+     m_mainWindow = ApplicationInfo::getWindow()->findChild<QObject*>("menu");
+     QObject::connect(m_mainWindow,SIGNAL(textchanged(QString)),this,SLOT(display(QString)));
 
+}
+void ActivityInfoTree::display(QString text)
+{
+    int end,length,start = 0;
+
+
+
+       searchedtext = text;
+       end = m_menuTreeFulltemp.length();
+       length = text.length(); // length of the  string . i.e text in the textfield
+       m_menuTree.clear();
+
+       // templength = length of string that was there in the textfield before the present string
+       // if user erases a character from a string ,the search needs to be performed on all the activities
+       if(length < templength)
+       {
+
+           m_menuTreeFulltemp = m_menuTreeFull;
+           end = m_menuTreeFulltemp.length();
+       }
+
+       if(text.trimmed() == "")
+       {
+           m_menuTree.clear();
+           m_menuTreeFulltemp = m_menuTreeFull;
+           templength = 0;
+           emit menuTreeChanged();
+       }
+       else
+       {
+
+           while(start != end )
+           {
+               // if there is any activity that matches the description push it into the m_menutree
+               if(m_menuTreeFulltemp.at(start)->title().contains(text.trimmed(),Qt::CaseInsensitive))
```

Comment:
I agree with Holger. When doing support we need to be able to direct a
user to search the real name in english.

Le 25/01/2016 15:15, Holger Kaelberer a écrit :
>
> In src/core/ActivityInfoTree.cpp
> <https://github.com/bdoin/GCompris-qt/pull/110#discussion_r50697427>:
>
> > +       }
> > +
> > +       if(text.trimmed() == "")
> > +       {
> > +           m_menuTree.clear();
> > +           m_menuTreeFulltemp = m_menuTreeFull;
> > +           templength = 0;
> > +           emit menuTreeChanged();
> > +       }
> > +       else
> > +       {
> > +
> > +           while(start != end )
> > +           {
> > +               // if there is any activity that matches the description push it into the m_menutree
> > +               if(m_menuTreeFulltemp.at(start)->title().contains(text.trimmed(),Qt::CaseInsensitive))
>
> Tend to agree with rish9511: Title will probably be searched for most
> often. Not sure if also scanning description won't produce too many
> result. Might need to try it. Maybe configurable.
>
> As a developper I'd really love to see the 'internal' name
> (activityInfo.name.split('/')[0]) searched because I best know this
> one, which is also not subject to translation. But that might not be a
> valid requirement :-) Or maybe also configurable?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bdoin/GCompris-qt/pull/110/files#r50697427>.
>